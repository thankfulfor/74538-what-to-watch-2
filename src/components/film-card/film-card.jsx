import * as React from 'react';
import PropTypes from 'prop-types';
import {VideoPlayer} from '../video-player/video-player.jsx';

export const FilmCard = (props) => {
  const {film, isPlaying, onMouseEnter, onMouseLeave} = props;

  return (
    <article
      onMouseEnter={onMouseEnter}
      onMouseLeave={onMouseLeave}
      className="small-movie-card catalog__movies-card"
    >
      <div className="small-movie-card__image">
        {isPlaying
          ? (
            <VideoPlayer
              src={film.previewVideoLink}
              poster={film.previewImage}
              isPlaying={isPlaying}
            />)
          : (<img src={film.previewImage} alt={film.name} />)
        }
      </div>
      <h3 className="small-movie-card__title">
        <a className="small-movie-card__link" href="movie-page.html">{film.name}</a>
      </h3>
    </article>
  );
};

FilmCard.propTypes = {
  film: PropTypes.object.isRequired,
  isPlaying: PropTypes.bool.isRequired,
  onMouseEnter: PropTypes.func.isRequired,
  onMouseLeave: PropTypes.func.isRequired,
};
