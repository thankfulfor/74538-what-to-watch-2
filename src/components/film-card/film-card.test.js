import React from 'react';
import {FilmCard} from './film-card';
import {BrowserRouter} from 'react-router-dom';
import renderer from 'react-test-renderer';

const film = {
  previewImage: `img/we-need-to-talk-about-kevin.jpg`,
  name: `We need to talk about Kevin`,
  genre: `drama`,
  previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/b/b3/Big_Buck_Bunny_Trailer_400p.ogv`
};

it(`FilmCard корректно рендерится после перезапуска`, () => {
  const handleMouseMove = jest.fn();
  const handleClick = jest.fn();
  const tree = renderer.create(
      <BrowserRouter>
        <FilmCard
          onCardClick={handleClick}
          key={`${film.name}`}
          film={film}
          isPlaying={false}
          onMouseEnter={handleMouseMove}
          onMouseLeave={handleMouseMove}
        />
      </BrowserRouter>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
