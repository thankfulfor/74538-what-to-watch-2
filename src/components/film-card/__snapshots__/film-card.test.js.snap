// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmCard корректно рендерится после перезапуска 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onMouseEnter={[MockFunction]}
  onMouseLeave={[MockFunction]}
>
  <div
    className="small-movie-card__image"
  >
    <img
      alt="We need to talk about Kevin"
    />
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        We need to talk about Kevin
      </a>
    </h3>
  </div>
</article>
`;
